dandi/download.py

```diff
@@ -81,10 +101,38 @@ def download(
     else:
         raise ValueError(format)
 
+    if sync and not isinstance(parsed_url, SingleAssetURL):
+        client = parsed_url.get_client()
+        with client.session():
+            asset_paths = {asset["path"] for asset in parsed_url.get_assets(client)}
+        if isinstance(parsed_url, DandisetURL):
+            prefix = os.curdir
+            download_dir = output_path
+        elif isinstance(parsed_url, MultiAssetURL):
+            folder_path = op.normpath(parsed_url.path)
+            prefix = folder_path
+            download_dir = op.join(output_path, op.basename(folder_path))
+        else:
+            raise NotImplementedError(
+                f"Unexpected URL type {type(parsed_url).__name__}"
+            )
+        to_delete = []
+        for p in find_files(".*", download_dir, exclude_datalad=True):
+            if p == op.join(output_path, dandiset_metadata_file):
+                continue
+            a_path = op.normpath(op.join(prefix, op.relpath(p, download_dir)))
+            if on_windows:
+                a_path = a_path.replace("\\", "/")
+            if a_path not in asset_paths:
+                to_delete.append(p)
+        if to_delete and click.confirm(f"Delete {len(to_delete)} local assets?"):
```

Comment:
```suggestion
        if to_delete and click.confirm(f"Delete {len(to_delete)} local asset{'s' if len(to_delete) > 1 else ''}?"):
```

Reply:
Let's improve UX right away, and instead of plain confirm give a choice for "[l]ist" or alike and if asked, list all the assets to be deleted so user could make an informed decision

Reply:
filed related https://github.com/dandi/dandi-cli/issues/620 on allowing for non-interactive, which might be done after or melded into this PR if we agree before it is ready

Reply:
Done.